src/thunderbots/software/radio_communication/mrf_backend.cpp

```diff
@@ -50,3 +132,261 @@ void MRFBackend::update_dongle_events()
 {
     dongle.handle_libusb_events();
 }
+
+thunderbots_msgs::RobotStatus MRFBackend::handle_message(int index, const void *data,
+                                                         std::size_t len, uint8_t lqi,
+                                                         uint8_t rssi)
+{
+    thunderbots_msgs::RobotStatus robot_status;
+
+    robot_status.link_quality = lqi / 255.0;
+
+    {
+        bool found = false;
+        for (std::size_t i = 0; !found && i < sizeof(RSSI_TABLE) / sizeof(*RSSI_TABLE);
+             ++i)
+        {
+            if (RSSI_TABLE[i].rssi < rssi)
+            {
+                robot_status.received_signal_strength_db = RSSI_TABLE[i].db;
+                found                                    = true;
+            }
+        }
+        if (!found)
+        {
+            robot_status.received_signal_strength_db = -90;
+        }
+    }
+
+    const uint8_t *bptr = static_cast<const uint8_t *>(data);
```

Comment:
`bptr`?

Reply:
...don't know of any better name, that was part of the legacy code